pkg/agent/utils.go

```diff
@@ -473,8 +473,6 @@ func setCustomKubeletConfig(customKc *datamodel.CustomKubeletConfig,
 		}
 		if customKc.TopologyManagerPolicy != "" {
 			kubeletConfig.TopologyManagerPolicy = customKc.TopologyManagerPolicy
-			// enable TopologyManager feature gate is required for this configuration.
-			kubeletConfig.FeatureGates["TopologyManager"] = true
```

Comment:
sorry I think you clarified this earlier but I can't find the messages - are we sure RP is currently setting this feature flag accordingly? otherwise this could introduce problems

Reply:
ah, reread the PR description

so from reading this: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/, it seems like we haven't needed to set this feature gat since 1.18 since it was defaulted to true?

Reply:
We don't need to set "TopologyManager" in agent baker, since it is default true for 1.18+ clusters.  So this PR is safe to merge and has no release dependency for RP.

Another PR (#4229 ) is to remove CustomCPUCFSQuotaPeriod logic from agent baker, this is dependent on RP and I'll update the PR once we have RP updated.